### PR TITLE
weight parallel

### PR DESCRIPTION
**Pull request by chang-wenbin**

### CODE REVIEW

File: ppdiffusers/ppdiffusers/models/simplified_sd3.py [hunk 1]
```diff
@@ -31,14 +35,29 @@ def __init__(self, num_layers: int, dim: int, num_attention_heads: int, attentio
...
-        self.to_out_linear = nn.LayerList([nn.Linear(self.dim, self.dim) for i in range(num_layers)])
-        self.to_add_out_linear = nn.LayerList([nn.Linear(self.dim, self.dim) for i in range(num_layers - 1)])
-        self.ffn1 = nn.LayerList([nn.Linear(self.dim, self.dim * 4) for i in range(num_layers)])
-        self.ffn2 = nn.LayerList([nn.Linear(self.dim * 4, self.dim) for i in range(num_layers)])
-        self.ffn1_context = nn.LayerList([nn.Linear(self.dim, self.dim * 4) for i in range(num_layers - 1)])
-        self.ffn2_context = nn.LayerList([nn.Linear(self.dim * 4, self.dim) for i in range(num_layers - 1)])
+
+        
+        if model_parallel_size > 1:
+            self.qkv_mp = nn.LayerList([fleet.meta_parallel.ColumnParallelLinear(self.dim, 3 * self.dim, gather_output=False, has_bias=True) for i in range(num_layers)])
```
Comment (zhoutianzi666) [inline ppdiffusers/ppdiffusers/models/simplified_sd3.py]: from fleet.meta_parallel import RowParallelLinear
from fleet.meta_parallel import ColumnParallelLinear

被频繁使用了，应该预先import as吧

Comment (chang-wenbin) [reply ppdiffusers/ppdiffusers/models/simplified_sd3.py]: 收到，已修改～

Comment (zhoutianzi666) [reply ppdiffusers/ppdiffusers/models/simplified_sd3.py]: > from fleet.meta_parallel import RowParallelLinear
> from fleet.meta_parallel import ColumnParallelLinear

from fleet.meta_parallel import RowParallelLinear
from fleet.meta_parallel import ColumnParallelLinear
改成这样

File: ppdiffusers/ppdiffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3.py [hunk 2]
```diff
@@ -800,23 +799,24 @@ def __call__(
...
-                    pooled_prompt_embeds_input ,pooled_prompt_embeds_ = paddle.split(pooled_prompt_embeds,2,axis=0)
-                    
-                    dist.scatter(latent_input,[latent_input,latent_model_input_])
-                    dist.scatter(timestep_input,[timestep_input,timestep_])
-                    dist.scatter(prompt_embeds_input,[prompt_embeds_input,prompt_embeds_])
-                    dist.scatter(pooled_prompt_embeds_input,[pooled_prompt_embeds_input,pooled_prompt_embeds_])
+
+                hcg = fleet.get_hybrid_communicate_group()
+                dp_degree = hcg.get_data_parallel_world_size()
+                if dp_degree > 1 and self.do_classifier_free_guidance:
```
Comment (zhoutianzi666) [inline ppdiffusers/ppdiffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3.py]: `dp_degree > 1 and self.do_classifier_free_guidance:`这个判断出现了两次了！
这样不好！建议声明一个临时变量！

**Review by westfish — DISMISSED** [review]

**Review by westfish — APPROVED** [review]

**Review by jerrywgz — APPROVED** [review]

**Pull request merged**